include/git2/remote.h

```diff
@@ -474,6 +474,17 @@ typedef int GIT_CALLBACK(git_push_negotiation)(const git_push_update **updates,
  */
 typedef int GIT_CALLBACK(git_push_update_reference_cb)(const char *refname, const char *status, void *data);
 
+/**
+ * Callback to resolve URLs before connecting to remote
+ *
+ * @param url The URL to resolve
+ * @param url_resolved The resolved URL
```

Comment:
Could you please clarify that the resolved URL is expected to belong to the caller afterwards?

Reply:
Also probably worth explicitly calling out the case where we return GIT_EPASSTHROUGH here as a case where we don't change anything if you're adding that change too.

Reply:
I don't white understand the case for `GIT_EPASSTHROUGH`. If the callback is not set, the `url` is automatically copied to `resolved_url`. If the caller is set, the caller has to ensure `resolved_url` is set. Do you suggest that the callback can return `GIT_EPASSTHROUGH`, and the caller automatically `strdup`s it?

Reply:
Sorry, didn't see your comment. But yes, that's exactly it.